steps/cache-pull/2.2.0/step.yml

```diff
@@ -0,0 +1,103 @@
+summary: Download the build cache from bitrise.io.
+description: |-
+  Downloads the build cache from bitrise.io, and moves the files back to
+  their original place. The aim is to speed up your builds as not having to download every single dependency every time can be a serious timesaver.
+
+  ### Configuring the Step
+
+  To be able to pull from a previous build's cache, you need two things.
+
+  1. The **Cache:Push** Step at the end of your Workflow.
+  2. At least one successful build on the same branch in the last seven days.
+
+  If there was no successful build on the branch in question in the last seven days, the cache automatically gets deleted.
+
+  ### Troubleshooting
+
+  If the Step fails, check the time of the latest build. If there was a successful build in the last seven days, check if the Workflow included the **Cache:Push** Step.
+
+  If the **Cache:Push** Step was present, check the inputs of the Step. It's possible, for example, that the provided cache path is incorrect.
+
+  ### Useful links
+
+  - [Caching](https://devcenter.bitrise.io/builds/caching/about-caching-index/)
+
+  ### Related Steps
+
+  - [Cache:Push](https://www.bitrise.io/integrations/steps/cache-push)
+  - [Run CocoaPods install](https://www.bitrise.io/integrations/steps/cocoapods-install)
+  - [Run npm command](https://www.bitrise.io/integrations/steps/npm)
+  - [yarn](https://www.bitrise.io/integrations/steps/yarn)
+website: https://github.com/bitrise-steplib/steps-cache-pull
+source_code_url: https://github.com/bitrise-steplib/steps-cache-pull.git
+support_url: https://github.com/bitrise-steplib/steps-cache-pull/issues
+published_at: 2020-10-27T10:39:35.056615129Z
+source:
+  git: https://github.com/bitrise-steplib/steps-cache-pull.git
+  commit: 7b48d11cc95b594f14f75df29fc4aa7c7b5a7d95
+host_os_tags:
+- osx-10.10
+- ubuntu-14.04
+type_tags:
+- utility
+toolkit:
+  go:
+    package_name: github.com/bitrise-steplib/steps-cache-pull
+deps:
+  apt_get:
+  - name: tar
+is_requires_admin_user: false
+is_always_run: false
+is_skippable: true
+run_if: .IsCI
+inputs:
+- opts:
+    description: Working directory path - should be an absolute path.
+    summary: Working directory path
+    title: Working directory path
+  workdir: $BITRISE_SOURCE_DIR
+- cache_api_url: $BITRISE_CACHE_API_URL
+  opts:
+    description: Cache API URL
+    is_dont_change_value: true
+    summary: Cache API URL
+    title: Cache API URL
+- is_debug_mode: "false"
+  opts:
+    description: If debug mode is enabled, the step will print verbose logs
+    is_required: true
+    summary: If debug mode is enabled, the step will print verbose logs
+    title: Debug mode?
+    value_options:
+    - "true"
+    - "false"
+- is_direct_url: "false"
+  opts:
+    category: Debug
+    description: Is the provided URL direct download address?
+    is_dont_change_value: true
+    is_required: true
+    title: Sidestep Cache API?
+    value_options:
+    - "true"
+    - "false"
```

Comment:
can you please briefly explain the use of this input? what happens if some url is not direct?

Reply:
The `cache_api_url` input stores a URL from where we can ask Bitrise for the given build's cache download URL. If you turn on this input the step will directly download the content of the `cache_api_url`.

This input is supposed to be used for testing, that's why  `is_dont_change_value` option is set.

Do you think we should update the input description?

Reply:
What happens if this turned off? 🤔 Could users use this for testing or only us? In the latter case I'd suggest not to add this input here at all.

Reply:
Turned off by default, in this case, the step behaves as previously: `cache_api_url` can be used to [get the build's cache archive download URL](https://github.com/bitrise-steplib/steps-cache-pull/blob/master/main.go#L204-L209) (that's why it is indirect, the cache_api_url is not the location of the cache archive).

Anyone can use this option (in bitrise.yml mode): let's say you store your cache archive on GC Storage: you turn on `is_direct_url` and set the `cache_api_url` to the GC Storage URL, in this case, the step will directly use the cache_api_url to download the cache archive. 

We introduced this input to be able to test the step outside of a Bitrise build and be able to test the step with different test archives: https://github.com/bitrise-steplib/steps-cache-pull/blob/master/bitrise.yml#L254-L277